app/components/RemixLink.tsx

```diff
@@ -0,0 +1,60 @@
+import { useHref } from "@remix-run/react";
+import type { LinkProps } from "@remix-run/react";
+import React, { forwardRef } from "react";
+import type { Ref } from "react";
+import { useLinkClickHandler } from "react-router-dom";
+import { Button, Link } from "@navikt/ds-react";
+
+interface IProps extends LinkProps {
+  as: "Link" | "Button";
+  children: React.ReactElement | string;
+}
+
+export const RemixLink = forwardRef(RemixLinkComponent);
+
+function RemixLinkComponent(
+  { onClick, replace = false, state, target, to, as, children }: IProps,
+  ref: Ref<HTMLAnchorElement> | undefined
+) {
+  const href = useHref(to);
+  const handleClick = useLinkClickHandler(to, {
+    replace,
+    state,
+    target,
+  });
+
+  if (as === "Button") {
+    return (
+      <Button
+        href={href}
+        onClick={(event) => {
+          // @ts-ignore
+          onClick?.(event);
+          if (!event.defaultPrevented) {
+            // @ts-ignore
+            handleClick(event);
+          }
+        }}
+        ref={ref}
+        as="a"
+      >
+        {children}
+      </Button>
+    );
+  }
+
+  return (
+    <Link
+      href={href}
+      onClick={(event) => {
+        onClick?.(event);
+        if (!event.defaultPrevented) {
```

Comment:
Jeg skjønner ikke helt hva som skjer her med defaultPrevented. Man fyrer alltid av onClick hvis man sender det inn som props. Og hvis ikke event er defaultPrevented (hvor kommer det i så fall fra?) så fyrer vi av remix sin click handler. Hvorfor fyrer vi ikke alltid av remix sin click handler?

Reply:
Godt spørsmål! 💯 

Remix sin click-handler (`handleClick` i vår kode) er den som navigerer internt i applikasjonen uten nødvendigvis å kjøre en full refresh. Den vil vi skal kjøre etter brukerens custom onClick-funksjon, hvis denne finnes.

Hvis man har sendt inn en custom onClick-handler skal denne alltid kjøre. Her får man inn en event, og den kan man kjøre `preventDefault()` på, for å kansellere alt som skulle trigges videre. Da skal ikke `handleClick` kjøres. Slik:

```
function eventHandler(e:Event) {
  // Gjør noe kult her, vi vil ikke navigere videre på lenka når den klikkes på
  e.preventDefault();
}

<RemixLink onClick={eventHandler} to="url-her">Lenketekst</RemixLink>
```

Jeg holdt logikken nøyaktig slik den ligger i [react-router](https://github.com/remix-run/react-router/blob/main/packages/react-router-dom/index.tsx#L469). Ser også at remix [legger på prefetching ](https://github.com/remix-run/remix/blob/main/packages/remix-react/components.tsx#L314)på sin variant av `<Link>`, noe vi også kan fikse på i fremtiden.